### ISSUE TITLE
Windows in a BTRFS subvolume

### ISSUE THREAD

**Issue by Anuskuss** [maharmstone/quibble#102]
Can this work?
1. Install [WinBtrfs](https://github.com/maharmstone/btrfs)
2. Setup [Quibble](https://github.com/maharmstone/quibble) and point it to Linux partition (`freeldr.ini`)
3. Turn Windows NTFS partition into BTRFS
```shell
ntfs2btrfs /dev/nvme0n1p4
```
4. Send Windows partition to Linux
```shell
mount /dev/nvme0n1p4 /mnt
btrfs subvolume snapshot -r /mnt /mnt/@windowsro
btrfs send /mnt/@windowsro | btrfs receive /
btrfs subvolume snapshot /@windowsro /@windows
btrfs subvolume delete /@windowsro
```
5. Remove Windows partition and resize Linux partition
```shell
fdisk /dev/nvme0n1
```
6. Point WinBtrfs to subvolume
```shell
cd /mnt/Windows/System32/config
chntpw -e SYSTEM
cd CurrentControlSet\Services\btrfs
cd $UUID
nv b SubvolId
ed SubvolId
```

**Comment by maharmstone** [comment]
Yes - see `Can I boot Btrfs from an arbitrary subvolume, like I can on Linux?` in the README.

**Comment by Anuskuss** [comment]
Okay, so I tried it with NTFS first and hit a few roadblocks:
```
File->Read returned EFI_UNSUPPORTED
Loading of WdFilter.sys failed
PE load returned EFI_UNSUPPORTED
load_image returned EFI_UNSUPPORTED
boot returned EFI_UNSUPPORTED
```

So I prevented `WdFilter.sys` (and `WdBoot.sys`) from loading:
```
sudo mount /dev/nvmen1p3 /mnt
cd /mnt/Windows/System32/config
chntpw -e SYSTEM
cd ControlSet001\Services\WdFilter
ed Start
0x04
```

But then I hit `UNMOUNTABLE_BOOT_VOLUME`. Any idea what to do now? I already disable Windows Defender. My OS is Windows 10 Enterprise LTSC 2021 (21H2).

**Comment by maharmstone** [comment]
```
File->Read returned EFI_UNSUPPORTED
Loading of WdFilter.sys failed
```

That's unexpected. Are you using the most recent version of Quibble?

**Comment by Anuskuss** [comment]
I'll retry with [master](https://github.com/maharmstone/quibble/tree/master) instead of [20230328](https://github.com/maharmstone/quibble/tree/20230328).

**Comment by Anuskuss** [comment]
It's a PITA to compile (can't find `efi.h` in `ntfs/src/ntfs.cpp`). Could you post a binary please?

Also tried to `cp -a` the NTFS files into a BTRFS subvolume but that just results in `INTERNAL_POWER_ERROR` (hibernate was disabled ofc).

**Comment by SzczurekYT** [comment]
> It's a PITA to compile (can't find `efi.h` in `ntfs/src/ntfs.cpp`). Could you post a binary please?
> 
> Also tried to `cp -a` the NTFS files into a BTRFS subvolume but that just results in `INTERNAL_POWER_ERROR` (hibernate was disabled ofc).

If you are compiling this on linux you will need to install a package that provides this header.
On fedora it's called `gnu-efi-devel`, you will need to find it's equivalent name for your distro.

**Comment by Anuskuss** [comment]
> you will need to install a package that provides this header.

I already got past that because `efi.h` does exist in my system (`/usr/lib/modules/6.6.0-arch1-1/build/include/linux/efi.h`) but including that (and a bunch of other locations) just results in `-fpermissive` errors. However installing `gnu-efi` does make it work so thanks for the tip!

Anyway nothing changed.

**Comment by maharmstone** [comment]
I didn't mean compile it yourself, just make sure that you're using 20230328!

Let me try with LTSC.

**Comment by maharmstone** [comment]
Okay, the issue was that in Release mode GCC was assuming that our copy of `memcpy` was returning a `void*` rather than `void`. Presumably this was undefined behaviour(?).

Please try overwriting drivers/ntfs.efi with https://github.com/maharmstone/ntfs-efi/releases/download/20231107/ntfs-amd64.efi.

**Comment by Anuskuss** [comment]
That doesn't help unfortunately. Also doesn't explain why BTRFS doesn't work either. If I had to take a guess I'd say that quibble assumes something (e.g. a file existing) which is untrue in LTSC. Or maybe there's some extra protection since this it is targeted towards businesses after all.

**Comment by Gustavoppw** [comment]
> Okay, so I tried it with NTFS first and hit a few roadblocks:
> 
> ```
> File->Read returned EFI_UNSUPPORTED
> Loading of WdFilter.sys failed
> PE load returned EFI_UNSUPPORTED
> load_image returned EFI_UNSUPPORTED
> boot returned EFI_UNSUPPORTED
> ```
> 
> So I prevented `WdFilter.sys` (and `WdBoot.sys`) from loading:
> 
> ```
> sudo mount /dev/nvmen1p3 /mnt
> cd /mnt/Windows/System32/config
> chntpw -e SYSTEM
> cd ControlSet001\Services\WdFilter
> ed Start
> 0x04
> ```
> 
> But then I hit `UNMOUNTABLE_BOOT_VOLUME`. Any idea what to do now? I already disable Windows Defender. My OS is Windows 10 Enterprise LTSC 2021 (21H2).

same issue as me but im on real hardware(and didnt try preventing those from booting on my regs yet) and qemu freezes before this point (10.1.4) for me
